[Issue] Build failure

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Forums discussion

_No response_

### Version

0.21 (Development)

### Full version info

```shell
macOS, Qt6
```


### Subproject(s) affected?

None

### Issue description

```
[3158/3866] Building CXX object src/Gui/CMakeFiles/FreeCADGui.dir/NotificationBox.cpp.o
FAILED: src/Gui/CMakeFiles/FreeCADGui.dir/NotificationBox.cpp.o 

/Users/bernie/CLionProjects/FreeCAD/src/Gui/NotificationBox.cpp:183:9: error: no member named 'init' in 'QStyleOptionFrame'
    opt.init(this);
/Users/bernie/CLionProjects/FreeCAD/src/Gui/NotificationBox.cpp:194:12: error: no member named 'init' in 'QStyleOption'
    option.init(this);
    ~~~~~~ ^
/Users/bernie/CLionProjects/FreeCAD/src/Gui/NotificationBox.cpp:339:50: warning: must specify at least one argument for '...' parameter of variadic macro [-Wgnu-zero-variadic-macro-arguments]
Q_GLOBAL_STATIC(QPalette, notificationbox_palette)
                                                 ^
/usr/local/lib/QtCore.framework/Headers/qglobalstatic.h:115:9: note: macro 'Q_GLOBAL_STATIC' defined here
#define Q_GLOBAL_STATIC(TYPE, NAME, ...)                                    \
        ^
1 warning and 2 errors generated.

```

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/src/Gui/NotificationBox.cpp
+++ b/src/Gui/NotificationBox.cpp
@@ -180,7 +180,7 @@ void NotificationLabel::paintEvent(QPaintEvent *ev)
 {
     QStylePainter p(this);
     QStyleOptionFrame opt;
-    opt.init(this);
+    opt.initFrom(this);
     p.drawPrimitive(QStyle::PE_PanelTipLabel, opt);
     p.end();
     QLabel::paintEvent(ev);
@@ -191,7 +191,7 @@ void NotificationLabel::resizeEvent(QResizeEvent *e)
     QStyleHintReturnMask frameMask;
     QStyleOption option;
 
-    option.init(this);
+    option.initFrom(this);
 
     if (style()->styleHint(QStyle::SH_ToolTip_Mask, &option, this, &frameMask)) {
         setMask(frameMask.region);

